modules/standard/IO.chpl

```diff
@@ -5607,19 +5607,20 @@ proc channel._format_reader(
         } else {
           // allocate regexp info if needed
           if r == nil then r = new unmanaged _channel_regexp_info();
```

Comment:
This line modifies the argument `r` so it needs ref intent does it not?

Reply:
Right. This is within _format_reader, where `r` remains by ref intent.
The method where I switched from ref to default intent is _match_regexp_if_needed().
(I still have hard times telling one from the other.)